### DISCUSSION TITLE
How to export audioBuffer with specified speed?

### DISCUSSION THREAD

**Discussion by UchihaYuki** [katspaugh/wavesurfer.js#3524] [Q&A] [answered]
I can change playback speed with:
```
this.generatedWaveSurfer.setPlaybackRate(this.speed, true);
```
But the audioBuffer exported by:
```
const generatedAudioBuffer = this.generatedWaveSurfer.getDecodedData();
```
Still maintains the original speed. 

I want to export a wave file with a specified speed and preserve pitch.

**Comment by katspaugh** [comment, marked as answer]
Yes, `getDecodedData` returns the original low-sample-rate decoded audio used for the rendering. Changing the playback rate doesn't affect it. If you want to speed up the audio, you'd have to resample the audio yourself.

Generally, wavesurfer.js doesn't aim to be a wrapper for all things Web Audio. It's just a visualization + player. It does allow connecting itself to a Web Audio graph by exporting its audio element (see [this example](https://wavesurfer.xyz/examples/?4436ec40a2ab943243755e659ae32196)) but nothign more than that. So please don't expect wavesurfer to cut, add effects, or process your audio in any other way.

**Reply by UchihaYuki** [reply]
I know it is not related to wavesurfer.js. But do you know how to change speed and preserve pitch at the same time while using Web Audio API. I found a related answer here:
https://stackoverflow.com/a/36940850/10974338
But it will change pitch. I think it should be pretty easy to solve since wavesurfer.js can preserve pitch easily. But I searched through the whole internet and haven't found the answer.

**Reply by katspaugh** [reply]
wavesurfer.js preserves the pitch because the audio element [does natively](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/preservesPitch).

It's not trivial to resample an audio buffer.
